create_app.py

```diff
@@ -105,9 +105,12 @@ def after_request(response):
         return response
 
     # TODO: Replace with an internal pagination mechanism
-    def calculate_total_pages(items_count: int) -> int:
+    def calculate_total_pages(items_count: Optional[int]) -> int:
         """Caculates the number of pages for the query"""
-        return math.ceil(items_count / ITEMS_PER_PAGE)
+        if items_count:
+            return math.ceil(items_count / ITEMS_PER_PAGE)
```

Comment:
You can use the yhe python integer division operator `//` instead of math.ceil here, although you'll probably need a + 1 to get the same effect

Reply:
Absolutely didn't think about this... but it's definitely much cleaner, fixed!

Reply:
The only thing we didn't think about is that if the division returns a whole number we shouldn't add +1 to it... guess we'll have to switch it back, even though it's not as clean...

Reply:
you mean this bit? 🤣 
> ...although you'll probably need a + 1 to get the same effect

Reply:
Yeah, except we need to not do it if it's a whole number because then it divides correctly, so there's no need to add the +1... Been trying to think of what check we could have for that case. Only just now realised we could probably do
```python
if items_count % ITEMS_PER_PAGE == 0:
    return items_count // ITEMS_PER_PAGE
else:
    return items_count // ITEMS_PER_PAGE + 1
```
Don't know if that's clearer though?

Reply:
yeah, fair enough, hadn't thought that through. Stick with math.ceil

Reply:
To be fair, I haven't thought about that either, I only realised it when tests failed after 68f0e0c 😂